Knossos.NET/Classes/Wine.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using Knossos.NET.Models;
+using System.Threading.Tasks;
+using System.Diagnostics;
+using System.IO;
+using System.ComponentModel;
+
+namespace Knossos.NET.Classes
+{
+    public struct WineResult
+    {
+        public bool IsSuccess;
+        public string ErrorMessage;
+
+        public WineResult(bool isSuccess, string errorMessage)
+        {
+            IsSuccess = isSuccess;
+            ErrorMessage = errorMessage;
+        }
+
+        public WineResult(bool isSuccess)
+        {
+            IsSuccess = isSuccess;
+            ErrorMessage = string.Empty;
+        }
+    }
+
+    /// <summary>
+    /// Small helper class to run Windows executables on Linux with Wine
+    /// </summary>
+    public static class Wine
+    {
+        public static async Task<WineResult> RunFred2(string exePath, string exeCmdLine, string? workingFolder, FsoExecArch fsoArch)
+        {
+            try
+            {
+                var wineArch = GetWineArch(fsoArch);
+                if (wineArch == null)
+                {
+                    return new WineResult(false, "Unsupported WINEARCH: " + fsoArch.ToString());
+                }
+
+                var prefixResult = await SetWinePrefixFred2(wineArch);
```

Comment:
Should this really be called every time you launch FRED?
Could this not early exit if the requested WINEPREFIX already exists or something?
I would imagine that the wineboot is pretty much a no-op if the prefix exists, so it wouldn't be too much of an issue to keep it as is though.

Reply:
I dont know how to detect if it completed successfully previusly what if it did error out? Running it again does nothing so i made it run every time.

Reply:
I see, makes sense. 
(purely in theory, this wineboot isn't necessary, just running wine on a new prefix works fine, but that doesn't set the DLL-override and will make the mono prompt show)